lib/handlers/quote/handler.ts

```diff
@@ -318,7 +320,23 @@ export async function getQuotes(quoterByRoutingType: QuoterByRoutingType, reques
       return res;
     })
   );
-  return quotes.filter((q): q is Quote => !!q);
+
+  const quotes: Quote[] = (
+    results.filter(
+      (result) => result.status == 'fulfilled' && result?.value !== null
+    ) as PromiseFulfilledResult<Quote | null>[]
+  ).map((result) => result.value as Quote);
+
+  const errors = results.filter(
+    (result) => result.status == 'rejected' && result?.reason.status >= 500
+  ) as PromiseRejectedResult[];
+
+  // throw QuoteError if there are no available quotes and at least one 5xx error
+  if (quotes.length === 0 && errors.length > 0) {
+    throw new QuoterError(errors[0].reason.message);
```

Comment:
safe to take errors[0]? is it possible both rfqQuoter + routingApiQuoter throw 500

Reply:
Right now it's not possible because the rfqQuoter suppresses any 500s. So this will be the routingApi error message. Not sure exactly how'd we want to surface two different error messages or how we choose which one to surface. Feel like for this purpose of just wanting 500s to returned for retries this is okay. Maybe I'll add a log that logs all the errors so we can get them during debugging.